EssentialCSharp.Web/Services/SiteMappingService.cs

```diff
@@ -5,6 +5,7 @@ namespace EssentialCSharp.Web.Services
     public class SiteMappingService : ISiteMappingService
     {
         public IList<SiteMapping> SiteMappings { get; }
+        public SiteMapping? CurrentSiteMapping { get; set; }
```

Comment:
This does not look to be the right place to put this information. We should be passing this directly through to the view not storing it in a service.

Reply:
Adjusted to use ViewBag